pennylane/plugins/expt_tensornet.py

```diff
@@ -246,6 +228,43 @@ def var(self, observable, wires, par):
 
         return self.ev(obs_nodes_for_squares, wires) - self.ev(obs_nodes, wires)**2
 
+    def sample(self, observable, wires, par):
+
+        if not isinstance(observable, list):
+            observable, wires, par = [observable], [wires], [par]
+
+        matrices = [self._get_operator_matrix(o, p) for o, p in zip(observable, par)]
+
+        decompositions = [spectral_decomposition(A) for A in matrices]
+        eigenvalues, projector_groups = list(zip(*decompositions))
+        eigenvalues = list(eigenvalues)
+
+        # Matching each projector with the wires it acts on
+        # while preserving the groupings
+        projectors_with_wires = [[(proj, wires[idx]) for proj in proj_group]
+                                 for idx, proj_group in enumerate(projector_groups)]
+
+        # The eigenvalue - projector maps are preserved as product() preserves
+        # the previous ordering by creating a lexicographic ordering
+        joint_outcomes = list(product(*eigenvalues))
+        projector_tensor_products = list(product(*projectors_with_wires))
+
+        joint_probabilities = [] #[(outcome, 0.0) for idx, outcome in enumerate(joint_outcomes)]
+
+        for projs in projector_tensor_products:
+            obs_nodes = []
+            obs_wires = []
+            for proj, wire in projs:
+
+                tensor = proj.reshape([2] * len(wire) * 2)
+                obs_nodes.append(self._add_node(tensor, wire))
+                obs_wires.append(wire)
+
+            joint_probabilities.append(self.ev(obs_nodes, obs_wires))
+
+        outcomes = np.array([np.prod(p) for p in joint_outcomes])
```

Comment:
Nice, so this will allow us to update to tuples pretty trivially, correct? (essentially just delete this line?)

Reply:
I believe so, yes